proposed/2021/NuGetConsumePDBInPackageReference.md

```diff
@@ -0,0 +1,212 @@
+
+
+## Open Questions
+
+* In which version will this functionality be ready?
+
+* When there is no other extensions, shall NuGet provide empty string `"related": ""`, or just skip adding `"related": ""` in assets file?
+
+* Shall NuGet list not only .pdb and .xml, but also all the extensions in [AllowedReferenceRelatedFileExtensions](https://github.com/dotnet/msbuild/blame/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L621-L627) if there is any? (.pdb, .xml, .pri, .dll.config, .exe.config)
+
+
+## Considerations
+
+### 1. If there is a change in `related` property, will it make any difference in restore? 
+For a package reference, there should be no change in `related` property unless we change to reference a different package/version. So there is no difference in restore when having `related` property.
+
+### 2. In addition to the proposed approach, 2 of other solutions were considered. 
+
+#### **Do nothing - Recommend the customers use the custom target workaround**
+
+The workaround that copying .pdb and .xml file into build output folder(.NET Framework) is as following:
+```xml
+ <Target Name="_ResolveCopyLocalNuGetPackagePdbsAndXml" 
+          Condition="$(CopyLocalLockFileAssemblies) == true" 
+          AfterTargets="ResolveReferences">
+    <ItemGroup>
+      <ReferenceCopyLocalPaths 
+        Include="@(ReferenceCopyLocalPaths->'%(RootDir)%(Directory)%(Filename).pdb')" 
+        Condition="'%(ReferenceCopyLocalPaths.NuGetPackageId)' != ''
+                    and Exists('%(RootDir)%(Directory)%(Filename).pdb')" />
+      <ReferenceCopyLocalPaths 
+        Include="@(ReferenceCopyLocalPaths->'%(RootDir)%(Directory)%(Filename).xml')" 
+        Condition="'%(ReferenceCopyLocalPaths.NuGetPackageId)' != ''
+                    and Exists('%(RootDir)%(Directory)%(Filename).xml')" />
+    </ItemGroup>
+  </Target>
+```
+Pros:
+* It works (in certain condition).
+
+Cons:
+* It only works for .NET Framework. For .NET Core, it needs to be adjusted.
+* It only copies .pdb and .xml file to build output. If publish output path is different from build output path, it needs to be adjusted to publish .pdb and .xml.
+
+
+#### **Add as LockFileItem, but not a property of LockFileItem**
+
+Instead of adding existing extension as a property of LockFileItem, adding each exsisting file as a property as a LockFileItem.
+So the targets section in assets file is as following:
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {}
+        },
+        "runtime": {
+           "lib/netstandard2.0/PackageA.dll": {}
+        },
+        "symbol": {
+           "lib/netstandard2.0/PackageA.pdb": {}
+        },
+        "xml": {
+           "lib/netstandard2.0/PackageA.xml": {}
+        },
+    }
+  },
+```
+Pros:
+* It's as powerful as adding existing extension as a property of LockFileItem.
+
+Cons:
```

Comment:
In the cons, I'd add that `symbol` or `xml` are not really on the same level as `compile` or `runtime` items. 
The usage of these files are SDK driven rather than package author driven.

Reply:
Thanks for the great point! Added.